core/app/models/spree/tax/order_adjuster.rb

```diff
@@ -16,7 +16,9 @@ def initialize(order)
       def adjust!
         return unless order_tax_zone(order)
 
-        order.all_adjustments.tax.destroy_all
+        [order, *order.line_items, *order.shipments].each do |item|
+          item.adjustments.destroy(item.adjustments.select(&:tax?))
+        end
```

Comment:
Instead of this we should probably fix this line https://github.com/solidusio/solidus/blob/master/core/app/models/spree/tax/item_adjuster.rb#L31

and then remove the `skip_destroy_adjustments` option